src/Microsoft.CSharp/src/Microsoft/CSharp/RuntimeBinder/Syntax/NameManager.cs

```diff
@@ -2,35 +2,168 @@
+            new Name("Concat"),
+            new Name("Add"),
+            new Name("get_Length"),
+            new Name("get_Chars"),
+            new Name("CreateDelegate"),
+            new Name("FixedElementField"),
+            new Name("HasValue"),
+            new Name("get_HasValue"),
+            new Name("Value"),
+            new Name("get_Value"),
+            new Name("GetValueOrDefault"),
+            new Name("?"),
+            new Name("<?>"),
+            new Name("Lambda"),
+            new Name("Parameter"),
+            new Name("Constant"),
+            new Name("Convert"),
+            new Name("ConvertChecked"),
+            new Name("AddChecked"),
+            new Name("Divide"),
+            new Name("Modulo"),
+            new Name("Multiply"),
+            new Name("MultiplyChecked"),
+            new Name("Subtract"),
+            new Name("SubtractChecked"),
+            new Name("And"),
+            new Name("Or"),
+            new Name("ExclusiveOr"),
+            new Name("LeftShift"),
+            new Name("RightShift"),
+            new Name("AndAlso"),
+            new Name("OrElse"),
+            new Name("Equal"),
+            new Name("NotEqual"),
+            new Name("GreaterThanOrEqual"),
+            new Name("GreaterThan"),
+            new Name("LessThan"),
+            new Name("LessThanOrEqual"),
+            new Name("ArrayIndex"),
+            new Name("Assign"),
+            new Name("Condition"),
+            new Name("Field"),
+            new Name("Call"),
+            new Name("New"),
+            new Name("Quote"),
+            new Name("ArrayLength"),
+            new Name("UnaryPlus"),
+            new Name("Negate"),
+            new Name("NegateChecked"),
+            new Name("Not"),
+            new Name("NewArrayInit"),
+            new Name("Property"),
+            new Name("AddEventHandler"),
+            new Name("RemoveEventHandler"),
+            new Name("InvocationList"),
+            new Name("GetOrCreateEventRegistrationTokenTable"),
+            /* Above here corresponds with PredefinedName enum */
+            new Name("true"),
+            new Name("false"),
+            new Name("null"),
+            new Name("base"),
+            new Name("this"),
+            new Name("explicit"),
+            new Name("implicit"),
+            new Name("__arglist"),
+            new Name("__makeref"),
+            new Name("__reftype"),
+            new Name("__refvalue"),
+            new Name("as"),
+            new Name("checked"),
+            new Name("is"),
+            new Name("typeof"),
+            new Name("unchecked"),
+            new Name("void"),
+            new Name("")
+        };
 
-        internal NameManager()
-            : this(new NameTable())
-        {
-        }
+        private static readonly NameTable s_knownNames = GetKnownNames();
 
-        private NameManager(NameTable nameTable)
+        private static NameTable GetKnownNames()
         {
```

Comment:
I think it is ok. Some runtimes (like mono) used to have problems with complicated scenarios involving static ctors, but this one looks fairly simple.

It may be worth making s_knownNames a string switch though.

---
In reply to: [108529374](https://github.com/dotnet/corefx/pull/17605#discussion_r108529374) [](ancestors = 108529374)